$OpenBSD: patch-CMakeLists_txt,v 1.2 2019/01/30 18:55:47 cwen Exp $

Don't use -Wl,--no-undefined on OpenBSD.  This causes linker errors
because OpenBSD doesn't link shared objects to libc, so libc symbols
are undefined.  Even if we add libc to the link, ports-gcc errors,
"undefined reference to `pthread_cancel'", because libgcc.a refers to
pthread_cancel, a symbol in libpthread or weak symbol in libestdc++.

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -62,7 +62,7 @@ if (CMAKE_COMPILER_IS_GNUCXX)
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic-errors")
   endif()
 
-  if (NOT APPLE)
+  if (NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL OpenBSD)
     set ( CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined ${CMAKE_SHARED_LINKER_FLAGS}")
     set ( CMAKE_MODULE_LINKER_FLAGS "-Wl,--fatal-warnings -Wl,--no-undefined ${CMAKE_MODULE_LINKER_FLAGS}")
   endif ()
@@ -96,7 +96,7 @@ set (LIB_SUFFIX "" CACHE STRING "Define suffix of libr
 set( LIB_INSTALL_DIR lib${LIB_SUFFIX} )
 set( PLUGIN_INSTALL_DIR lib${LIB_SUFFIX}/grantlee/${Grantlee_MAJOR_MINOR_VERSION_STRING} )
 set( BIN_INSTALL_DIR bin )
-set( INCLUDE_INSTALL_DIR include )
+set( INCLUDE_INSTALL_DIR include/grantlee4 )
 set( DATA_INSTALL_DIR share/apps )
 
 # set up RPATH/install_name_dir
@@ -113,7 +113,7 @@ if (BUILD_GUI AND BUILD_TEXTDOCUMENT)
   add_subdirectory(textdocument)
 endif()
 
-set(CMAKECONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/grantlee" )
+set(CMAKECONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/grantlee4" )
 
 file(RELATIVE_PATH relInstallDir ${CMAKE_INSTALL_PREFIX}/${CMAKECONFIG_INSTALL_DIR} ${CMAKE_INSTALL_PREFIX} )
 
